File: core/cpp/src/test/vx/test_test.cpp
#include "../../main/vx/test.hpp"
#include "../../main/vx/core.hpp"
#include "../../main/vx/data/file.hpp"
#include "../../main/vx/web/html.hpp"
#include "test_test.hpp"

namespace vx_test_test {


  vx_test::Type_testcase f_file_test(vx_core::Type_context context) {
    vx_core::vx_log("Test Start: f_file_test");
    // testdescribe_1
    vx_test::Type_testresult testresult_1 = vx_test::f_test_true(
      context,
      vx_data_file::f_boolean_exists_from_file(
        vx_core::f_new(
          vx_data_file::t_file,
          vx_core::vx_new(vx_core::t_anylist, {
            vx_core::vx_new_string(":name"),
            vx_core::vx_new_string("testsuite.vxlisp"),
            vx_core::vx_new_string(":path"),
            vx_core::c_path_test_resources
          })
        )
      )
    );
    vx_test::Type_testdescribe testdescribe_1 = vx_core::vx_new(vx_test::t_testdescribe, {
      vx_core::vx_new_string(":describename"), vx_core::vx_new_string("(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuite.vxlisp\"\n   :path path-test-resources)))"),
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":testresult"), testresult_1
    });
    vx_core::vx_Type_listany listdescribe = {
      testdescribe_1
    };
    vx_test::Type_testcase output = vx_core::vx_new(vx_test::t_testcase, {
      vx_core::vx_new_string(":passfail"), vx_core::c_false,
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":casename"), vx_core::vx_new_string("file-test"),
      vx_core::vx_new_string(":describelist"),
      vx_core::vx_any_from_any(
        vx_test::t_testdescribelist,
        vx_test::t_testdescribelist->vx_new_from_list(listdescribe)
      )
    });
    vx_core::vx_log("Test End  : f_file_test");
    return output;
  }

  vx_test::Type_testcase f_file_testhtml(vx_core::Type_context context) {
    vx_core::vx_log("Test Start: f_file_testhtml");
    // testdescribe_1
    vx_test::Type_testresult testresult_1 = vx_test::f_test_true(
      context,
      vx_data_file::f_boolean_exists_from_file(
        vx_core::f_new(
          vx_data_file::t_file,
          vx_core::vx_new(vx_core::t_anylist, {
            vx_core::vx_new_string(":name"),
            vx_core::vx_new_string("testsuite.html"),
            vx_core::vx_new_string(":path"),
            vx_core::c_path_test_resources
          })
        )
      )
    );
    vx_test::Type_testdescribe testdescribe_1 = vx_core::vx_new(vx_test::t_testdescribe, {
      vx_core::vx_new_string(":describename"), vx_core::vx_new_string("(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuite.html\"\n   :path path-test-resources)))"),
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":testresult"), testresult_1
    });
    vx_core::vx_Type_listany listdescribe = {
      testdescribe_1
    };
    vx_test::Type_testcase output = vx_core::vx_new(vx_test::t_testcase, {
      vx_core::vx_new_string(":passfail"), vx_core::c_false,
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":casename"), vx_core::vx_new_string("file-testhtml"),
      vx_core::vx_new_string(":describelist"),
      vx_core::vx_any_from_any(
        vx_test::t_testdescribelist,
        vx_test::t_testdescribelist->vx_new_from_list(listdescribe)
      )
    });
    vx_core::vx_log("Test End  : f_file_testhtml");
    return output;
  }

  vx_test::Type_testcase f_file_testnode(vx_core::Type_context context) {
    vx_core::vx_log("Test Start: f_file_testnode");
    // testdescribe_1
    vx_test::Type_testresult testresult_1 = vx_test::f_test_true(
      context,
      vx_data_file::f_boolean_exists_from_file(
        vx_core::f_new(
          vx_data_file::t_file,
          vx_core::vx_new(vx_core::t_anylist, {
            vx_core::vx_new_string(":name"),
            vx_core::vx_new_string("testsuitenode.vxlisp"),
            vx_core::vx_new_string(":path"),
            vx_core::c_path_test_resources
          })
        )
      )
    );
    vx_test::Type_testdescribe testdescribe_1 = vx_core::vx_new(vx_test::t_testdescribe, {
      vx_core::vx_new_string(":describename"), vx_core::vx_new_string("(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuitenode.vxlisp\"\n   :path path-test-resources)))"),
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":testresult"), testresult_1
    });
    vx_core::vx_Type_listany listdescribe = {
      testdescribe_1
    };
    vx_test::Type_testcase output = vx_core::vx_new(vx_test::t_testcase, {
      vx_core::vx_new_string(":passfail"), vx_core::c_false,
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":casename"), vx_core::vx_new_string("file-testnode"),
      vx_core::vx_new_string(":describelist"),
      vx_core::vx_any_from_any(
        vx_test::t_testdescribelist,
        vx_test::t_testdescribelist->vx_new_from_list(listdescribe)
      )
    });
    vx_core::vx_log("Test End  : f_file_testnode");
    return output;
  }

  vx_test::Type_testcaselist test_cases(vx_core::Type_context context) {
    vx_core::vx_Type_listany listtestcase;
    listtestcase.push_back(vx_test_test::f_file_test(context));
    listtestcase.push_back(vx_test_test::f_file_testhtml(context));
    listtestcase.push_back(vx_test_test::f_file_testnode(context));
    vx_test::Type_testcaselist output = vx_core::vx_any_from_any(
      vx_test::t_testcaselist,
      vx_test::t_testcaselist->vx_new_from_list(listtestcase)
    );
    return output;
  }

  vx_test::Type_testcoveragesummary test_coveragesummary() {
    vx_test::Type_testcoveragesummary output = vx_core::vx_new(vx_test::t_testcoveragesummary, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":constnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(1)
      }),
      vx_core::vx_new_string(":docnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(93), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(43), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(46)
      }),
      vx_core::vx_new_string(":funcnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(10), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(3), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(28)
      }),
      vx_core::vx_new_string(":bigospacenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(34)
      }),
      vx_core::vx_new_string(":bigotimenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(34)
      }),
      vx_core::vx_new_string(":totalnums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(7), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(3), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(40)
      }),
      vx_core::vx_new_string(":typenums"), vx_core::vx_new(vx_test::t_testcoveragenums, {
        vx_core::vx_new_string(":pct"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":tests"), vx_core::vx_new_int(0), 
        vx_core::vx_new_string(":total"), vx_core::vx_new_int(11)
      })
    });
    return output;
  }

  vx_test::Type_testcoveragedetail test_coveragedetail() {
    vx_test::Type_testcoveragedetail output = vx_core::vx_new(vx_test::t_testcoveragedetail, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":typemap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":testcase"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testcaselist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testcoveragedetail"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testcoveragenums"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testcoveragesummary"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testdescribe"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testdescribelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testpackage"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testpackagelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testresult"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":testresultlist"), vx_core::vx_new_int(0)
      }),
      vx_core::vx_new_string(":constmap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":stylesheet-test"), vx_core::vx_new_int(0)
      }),
      vx_core::vx_new_string(":funcmap"), vx_core::vx_new(vx_core::t_intmap, {
        vx_core::vx_new_string(":context-test"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":div<-testcaselist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":div<-testpackage"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":div<-testpackagelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":divchildlist<-testpackagelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":file-test"), vx_core::vx_new_int(1),
        vx_core::vx_new_string(":file-testhtml"), vx_core::vx_new_int(1),
        vx_core::vx_new_string(":file-testnode"), vx_core::vx_new_int(1),
        vx_core::vx_new_string(":html<-divtest"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":p<-passfail"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":p<-testcoveragenums"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testcase"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testcaselist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testdescribe"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testdescribelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testpackage"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testpackagelist"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":resolve-testresult"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":security-test"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-false"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-false_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-gt"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-gt_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-ne"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-ne_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-string"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-string_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-true"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":test-true_1"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":tr<-testdescribe-casename"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":trlist<-testcase"), vx_core::vx_new_int(0),
        vx_core::vx_new_string(":trlist<-testcaselist"), vx_core::vx_new_int(0)
      })
    });
    return output;
  }

  vx_test::Type_testpackage test_package(vx_core::Type_context context) {
    vx_test::Type_testcaselist testcaselist = vx_test_test::test_cases(context);
    vx_test::Type_testcoveragesummary testcoveragesummary = vx_test_test::test_coveragesummary();
    vx_test::Type_testcoveragedetail testcoveragedetail = vx_test_test::test_coveragedetail();
    vx_test::Type_testpackage output = vx_core::vx_new(vx_test::t_testpackage, {
      vx_core::vx_new_string(":testpkg"), vx_core::vx_new_string("vx/test"),
      vx_core::vx_new_string(":caselist"), testcaselist,
      vx_core::vx_new_string(":coveragesummary"), testcoveragesummary,
      vx_core::vx_new_string(":coveragedetail"), testcoveragedetail
    });
    return output;
  }

}

File: core/csharp/test/AppTest/TestVx/TestTest.cs

namespace AppTest.TestVx;

public class TestTest {

  public static Vx.Test.Type_testcase f_file_test(Vx.Core.Type_context context) {
    Vx.Test.Type_testcase output = Vx.Core.vx_new(
      Vx.Test.t_testcase,
      ":passfail", false,
      ":testpkg", "vx/test",
      ":casename", "file-test",
      ":describelist",
      Vx.Core.vx_new(
        Vx.Test.t_testdescribelist,
        f_file_test_testdescribe_1(context)
      )
    );
    return output;
  }

  public static Vx.Test.Type_testdescribe f_file_test_testdescribe_1(Vx.Core.Type_context context) {
    Vx.Test.Type_testdescribe output = Vx.Core.vx_new(
      Vx.Test.t_testdescribe,
      ":describename", "(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuite.vxlisp\"\n   :path path-test-resources)))",
      ":testresult", Vx.Test.f_test_true(
        context,
        Vx.Data.File.f_boolean_exists_from_file(
          Vx.Core.f_new(
            Vx.Data.File.t_file,
            Vx.Core.vx_new(
              Vx.Core.t_anylist,
              Vx.Core.vx_new_string(":name"),
              Vx.Core.vx_new_string("testsuite.vxlisp"),
              Vx.Core.vx_new_string(":path"),
              Vx.Core.c_path_test_resources
            )
          )
        )
      )
    );
    return output;
  }

  public static Vx.Test.Type_testcase f_file_testhtml(Vx.Core.Type_context context) {
    Vx.Test.Type_testcase output = Vx.Core.vx_new(
      Vx.Test.t_testcase,
      ":passfail", false,
      ":testpkg", "vx/test",
      ":casename", "file-testhtml",
      ":describelist",
      Vx.Core.vx_new(
        Vx.Test.t_testdescribelist,
        f_file_testhtml_testdescribe_1(context)
      )
    );
    return output;
  }

  public static Vx.Test.Type_testdescribe f_file_testhtml_testdescribe_1(Vx.Core.Type_context context) {
    Vx.Test.Type_testdescribe output = Vx.Core.vx_new(
      Vx.Test.t_testdescribe,
      ":describename", "(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuite.html\"\n   :path path-test-resources)))",
      ":testresult", Vx.Test.f_test_true(
        context,
        Vx.Data.File.f_boolean_exists_from_file(
          Vx.Core.f_new(
            Vx.Data.File.t_file,
            Vx.Core.vx_new(
              Vx.Core.t_anylist,
              Vx.Core.vx_new_string(":name"),
              Vx.Core.vx_new_string("testsuite.html"),
              Vx.Core.vx_new_string(":path"),
              Vx.Core.c_path_test_resources
            )
          )
        )
      )
    );
    return output;
  }

  public static Vx.Test.Type_testcase f_file_testnode(Vx.Core.Type_context context) {
    Vx.Test.Type_testcase output = Vx.Core.vx_new(
      Vx.Test.t_testcase,
      ":passfail", false,
      ":testpkg", "vx/test",
      ":casename", "file-testnode",
      ":describelist",
      Vx.Core.vx_new(
        Vx.Test.t_testdescribelist,
        f_file_testnode_testdescribe_1(context)
      )
    );
    return output;
  }

  public static Vx.Test.Type_testdescribe f_file_testnode_testdescribe_1(Vx.Core.Type_context context) {
    Vx.Test.Type_testdescribe output = Vx.Core.vx_new(
      Vx.Test.t_testdescribe,
      ":describename", "(test-true\n (boolean-exists<-file\n  (file\n   :name \"testsuitenode.vxlisp\"\n   :path path-test-resources)))",
      ":testresult", Vx.Test.f_test_true(
        context,
        Vx.Data.File.f_boolean_exists_from_file(
          Vx.Core.f_new(
            Vx.Data.File.t_file,
            Vx.Core.vx_new(
              Vx.Core.t_anylist,
              Vx.Core.vx_new_string(":name"),
              Vx.Core.vx_new_string("testsuitenode.vxlisp"),
              Vx.Core.vx_new_string(":path"),
              Vx.Core.c_path_test_resources
            )
          )
        )
      )
    );
    return output;
  }

  public static Vx.Test.Type_testcaselist test_cases(Vx.Core.Type_context context) {
    object[] testcases = [
      TestTest.f_file_test(context),
      TestTest.f_file_testhtml(context),
      TestTest.f_file_testnode(context)
    ];
    Vx.Test.Type_testcaselist output = Vx.Core.vx_new(
      Vx.Test.t_testcaselist,
      testcases
    );
    return output;
  }

  public static Vx.Test.Type_testcoveragesummary test_coveragesummary() {
    Vx.Test.Type_testcoveragesummary output = Vx.Core.vx_new(
      Vx.Test.t_testcoveragesummary,
      ":testpkg", "vx/test", 
      ":constnums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 1), 
      ":docnums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 93, ":tests", 43, ":total", 46), 
      ":funcnums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 10, ":tests", 3, ":total", 28), 
      ":bigospacenums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 34), 
      ":bigotimenums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 34), 
      ":totalnums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 7, ":tests", 3, ":total", 40), 
      ":typenums", Vx.Core.vx_new(Vx.Test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 11)
    );
    return output;
  }

  public static Vx.Test.Type_testcoveragedetail test_coveragedetail() {
    Vx.Test.Type_testcoveragedetail output = Vx.Core.vx_new(
      Vx.Test.t_testcoveragedetail,
      ":testpkg", "vx/test",
      ":typemap", Vx.Core.vx_new(
        Vx.Core.t_intmap,
        ":testcase", 0,
        ":testcaselist", 0,
        ":testcoveragedetail", 0,
        ":testcoveragenums", 0,
        ":testcoveragesummary", 0,
        ":testdescribe", 0,
        ":testdescribelist", 0,
        ":testpackage", 0,
        ":testpackagelist", 0,
        ":testresult", 0,
        ":testresultlist", 0
      ), 
      ":constmap", Vx.Core.vx_new(
        Vx.Core.t_intmap,
        ":stylesheet-test", 0
      ), 
      ":funcmap", Vx.Core.vx_new(
        Vx.Core.t_intmap,
        ":context-test", 0,
        ":div<-testcaselist", 0,
        ":div<-testpackage", 0,
        ":div<-testpackagelist", 0,
        ":divchildlist<-testpackagelist", 0,
        ":file-test", 1,
        ":file-testhtml", 1,
        ":file-testnode", 1,
        ":html<-divtest", 0,
        ":p<-passfail", 0,
        ":p<-testcoveragenums", 0,
        ":resolve-testcase", 0,
        ":resolve-testcaselist", 0,
        ":resolve-testdescribe", 0,
        ":resolve-testdescribelist", 0,
        ":resolve-testpackage", 0,
        ":resolve-testpackagelist", 0,
        ":resolve-testresult", 0,
        ":security-test", 0,
        ":test", 0,
        ":test_1", 0,
        ":test-false", 0,
        ":test-false_1", 0,
        ":test-gt", 0,
        ":test-gt_1", 0,
        ":test-ne", 0,
        ":test-ne_1", 0,
        ":test-string", 0,
        ":test-string_1", 0,
        ":test-true", 0,
        ":test-true_1", 0,
        ":tr<-testdescribe-casename", 0,
        ":trlist<-testcase", 0,
        ":trlist<-testcaselist", 0
      )
    );
    return output;
  }

  public static Vx.Test.Type_testpackage test_package(Vx.Core.Type_context context) {
    Vx.Test.Type_testcaselist testcaselist = test_cases(context);
    Vx.Test.Type_testpackage output = Vx.Core.vx_new(
      Vx.Test.t_testpackage,
      ":testpkg", "vx/test", 
      ":caselist", testcaselist,
      ":coveragesummary", test_coveragesummary(),
      ":coveragedetail", test_coveragedetail()
    );
    return output;
  }

}
